pandas/core/internals/array_manager.py

```diff
@@ -352,12 +356,31 @@ def putmask(self, mask, new, align: bool = True):
             align_keys = ["mask"]
             new = extract_array(new, extract_numpy=True)
 
-        return self.apply_with_block(
-            "putmask",
-            align_keys=align_keys,
-            mask=mask,
-            new=new,
-        )
+        kwargs = {"mask": mask, "new": new}
```

Comment:
not feasible to go through 'apply'?

Reply:
Good point. So I forgot that `ArrayManager.apply` indeed already has this "align" logic as well. However, that is not used at the moment (only `putmask` and `where` define align keys, and those both still go through `apply_with_block` at the moment; this is also confirmed by codecov that it is unused).

But then I would rather remove it from `apply`, instead of going through `apply`, which will also prevent duplication. The reason for doing something custom: 1) `apply` is already quite complex (eg also dealing with ignoring failures), so being able to remove the "alignment" handling there would be nice (can still share this later with `where`), and 2) `putmask` can update the existing arrays, while `apply` always constructs a new manager, and 3) for the CoW branch I need to add copy-on-write logic to `putmask`, which is not needed for `apply` in general.